### ISSUE TITLE
Big blob scroll is laggy

### ISSUE THREAD

**Issue by valerybugakov** [sourcegraph/sourcegraph#29564]
#### Steps to reproduce:

1. Open https://sourcegraph.com/github.com/golang/go/-/blob/src/cmd/compile/internal/ssa/opGen.go?L34216
2. Scroll through the file. 

#### Expected behavior:

The scroll is smooth and doesn't cause reflow on every mouseover event.

#### Actual behavior:

The scroll is clunky and may crash the browser:

<img width="419" alt="Screen Shot 2022-01-11 at 12 35 16" src="https://user-images.githubusercontent.com/3846380/148936026-a012aa2d-5077-4ef2-8afb-1275930ec9d5.png">

## Time estimate

- Pull requests with ~450 lines changed should take **4 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
- Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

**Comment by umpox** [comment]
I wonder if this is related to our Tooltips, afaik they currently have a non-debounced `mouseover` event

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, let's spend 4 hours max on the root cause investigation.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov @umpox 
- I tried to disable `<Tooltip />` but it doesn't seem to help so I think `<Tooltip />` is not the root problem
- I see that the DOM size is very large so I tried to disable some parts of in `client/web/src/repo/blob/Blob.tsx` and found out that I don't move the mouse while scrolling or scroll with page up/down button, it seems to be better (still laggy because of DOM size), but if I scroll together with moving mouse, it becomes worse and worse. 

So for the first try I suggest to find a way to pause mouseover/mousemove observable until scroll is done. WDYT?
